src/dataframe/dataframe.jl

```diff
@@ -500,6 +500,57 @@ function Base.setindex!(df::DataFrame,
     return df
 end
 
+# df[SingleRowIndex, :] = Union{AbstractDict, NamedTuple}
+function Base.setindex!(df::DataFrame,
+                        row::Union{AbstractDict, NamedTuple},
```

Comment:
`DataFrameRow` isn't defined until after these functions are, how should I deal with that? I could move the definition of `DataFrameRow` up, or add a definition of `setindex!` after `DataFrameRow` gets defined. Both of those don't seem like great options to me, but I'm leaning towards adding a definition of `setindex!` in dataframerow.jl

Reply:
Alternatively I could take in an `AbstractDataFrame` and then verify the size, that might be best

Reply:
Yes, sounds like the best approach. BTW, there's a method for `DataFrame` above which doesn't look great (it uses the column order rather than names, and doesn't handle errors).

Reply:
So should I use column names strictly? I could see a few different use cases for `setindex!` where it takes in a dataframe, which gives us a few different options.

* We could just use column names and throw an error if every column name doesn't match
* We could check to see if the column names match, use them if they do, or just use column order if they don't
* We could only use column order

I'm not sure which one is really the behavior that people would expect (if someone is just thinking of the dataframe as a fancy matrix, then all they're worrying about is the dimensions), but I think the second option is the most forgiving of different expectations, so I'd think we should probably do that. If we do do that, it seems appropriate that we should also change the code for `df[MultiRowIndex, MultiColumnIndex] = DataFrame` around line 580 to have the same behavior. It seems like we should be able to expect that if we were to delegate one to the other that they'd give the same behavior

Reply:
What we do with `push!` and `vcat` is to use names and ignore order, so we should just do that here too for consistency.